packages/transit-vehicle-overlay/src/index.story.js

```diff
@@ -4,168 +4,287 @@ import { withA11y } from "@storybook/addon-a11y";
-          "routes/all?or=routes/90,100,190,200,290"
-        )}
-        tracked={text("block -or- trip id:", "track block: 9047")}
-        refreshDelay={number("refresh interval:", 5000)}
-        color={color("color:", "#777")}
-        highlightColor={color("tracked color:", "#ece90d")}
-        lowlightColor={color("trailing color:", "#AAA")}
-        recenterMap={boolean("follow vehicle:", true)}
-        panOffsetX={number("panOffsetX:", 0)}
-        panOffsetY={number("panOffsetY:", 0)}
-        hasPopup={boolean(
-          "use marker popups -- note: edit story to 'false' and refresh:",
-          true
-        )}
-        hasTooltip={boolean("use marker tooltips (desktop only):", true)}
-        onTrackedVehicleUpdate={console.log}
+    <BaseMap
+      center={PORTLAND}
+      zoom={INITIAL_ZOOM_LEVEL}
+      onViewportChanged={onViewportChanged}
+    >
+      <TransitVehicleOverlay
+        zoom={zoom}
+        center={center}
+        vehicleList={vehicleList}
+        onVehicleClicked={clickVehicle}
+        selectedVehicle={tv}
+        showOnlyTracked={showOnlyTracked}
+        pattern={getRoutePattern(tv)}
+        onRecenterMap={recenter}
+        color={clr}
+        highlightColor={highlightColor}
+        MarkerSlot={markers}
+        TooltipSlot={VehicleTooltip}
+        PopupSlot={VehiclePopup}
       />
     </BaseMap>
   );
 }
 
-function rtExampleLayerSwitcher() {
-  return rtExample("Real-Time Buses and Trains", true);
+function simple() {
+  // convert geojson data into expected pattern format
+  const pattern = utils.makePattern(geom, "111");
+  return simpleExample(all, pattern, "9050");
+}
+
+function alternate() {
+  const altVehicles = utils.convertAltData(altLine);
+  const altPattern = utils.makePattern(altGeom, "222");
+  return simpleExample(altVehicles, altPattern, "9088");
+}
+
+function simpleRectangles() {
+  return rectangles(false);
 }
-function rtExampleProgrammatic() {
-  return rtExample(null, false);
+
+function rtCircles() {
+  return realtimeExample(
+    proprietary.fetchVehicles,
+    proprietary.fetchPatternThrottled,
+    ModeCircles
+  );
+}
+
+function rtRectangles() {
+  return realtimeExample(
+    proprietary.fetchAltVehicles,
+    proprietary.fetchPatternThrottled,
+    ModeRectangles
+  );
 }
 
 storiesOf("TransitVehicleOverlay", module)
   .addDecorator(withA11y)
   .addDecorator(withInfo)
   .addDecorator(withKnobs)
-  .add("by Route", routeExample)
-  .add("all Routes", allExample)
-  .add("animated VehicleLayer", animatedExample)
-  .add("real-time vehicles in layer switcher", rtExampleLayerSwitcher)
-  .add("real-time vehicles programmatic control", rtExampleProgrammatic);
+  .add("simple overlay", simple)
+  .add("simple overlay (alternate vehicles service)", alternate)
+  .add("simple rectangles (click to select)", simpleRectangles)
+  .add("static rectangles (popups)", rectangles)
+  .add("real-time circles", rtCircles)
+  .add("real-time rectangles", rtRectangles);
```

Comment:
In general, with these stories it's helpful for me to see the actual component code as close as possible to the story declaration. I.e., 

```
  .add("simple overlay", () => {
    // convert geojson data into expected pattern format
    const patternGeometry = utils.makePattern(mockGeom, "111");
    // initial setup
    const fetchPattern = (vehicle, setter) => {
      utils.linterIgnoreTheseProps(vehicle);
      setter(patternGeometry);
    };
    // state setup for zoom (refreshes layer) and selected vehicles
    const [zoom, onViewportChanged] = utils.zoomState(INITIAL_ZOOM_LEVEL);
    const [
      getRoutePattern,
      getTrackedVehicle,
      updateTrackedVehicle
    ] = utils.trackedVehicleState(
      fetchPattern,
      utils.findVehicleById(allVehicles, "9050"), // initial vehicle_id
      patternGeometry
    );

    const [trackedVehicle, trackedRef] = getTrackedVehicle();
    utils.linterIgnoreTheseProps(trackedRef);
    VehiclePopup.defaultProps.setTracked = updateTrackedVehicle;
    return (
      <BaseMap
        center={PORTLAND}
        zoom={INITIAL_ZOOM_LEVEL}
        onViewportChanged={onViewportChanged}
      >
        <TransitVehicleOverlay
          zoom={zoom}
          vehicleList={allVehicles}
          onVehicleClicked={setClicked}
          selectedVehicle={trackedVehicle}
          pattern={getRoutePattern()}
          onRecenterMap={utils.recenterPanTo}
          // Imported components
          TooltipSlot={VehicleTooltip}
          PopupSlot={VehiclePopup}
          // Tip: Use storybook knobs to control the colors!
          color={color("color:", "#AAA")}
          highlightColor={color("isTracked color:", "#ece90d")}
          lowlightColor={color("trailing color:", "#AAA")}
        />
      </BaseMap>
    );
  })
```

There's probably a way to factor out some of the default props that are re-used by different stories, but I just generally find the usefulness of storybook in being able to click a story and see the code that represents that implementation. I know this might not be the case for some of the other otp-ui packages and that using predefined functions like you have might be somewhat unavoidable, but I do think we should strive for this.

Reply:
Your comment inspired me to create a simple -> complex stories, so a user can (hopefully) quickly get their head around the basic component.  So a good add, I think.

Placing a function between storybook .add and the implementation allowed me to demonstrate 6 different aspects of the component.  So keeping it DRY, there are 6 stories that show off various aspects (from using different service endpoints, to popups / no popups, etc...), but only 3 implementations of the component.